src/app/public/src/modules/table-of-contents/table-of-contents.component.ts

```diff
@@ -9,4 +10,53 @@ import { StacheNavLink } from '../nav';
 export class StacheTableOfContentsComponent {
   @Input()
   public routes: StacheNavLink[];
+  private viewTop: number;
+  private documentBottom: number;
+
+  constructor(
+    private windowRef: StacheWindowRef,
+    private omnibarService: StacheOmnibarAdapterService
+  ) {
+    this.windowRef.scrollEvent.subscribe(() => {
+      this.updateRoutesOnScroll(this.routes);
+    });
+  }
+
+  public updateRoutesOnScroll(routes: any[]) {
+    if (routes && routes.length) {
+      this.updateView(routes);
+    }
+  }
+
+  // Updates ng-class level trigger to show/hide blue border marking location on page
+  public updateView(routes: StacheNavLink[]) {
+    this.trackViewTop();
+    this.isCurrent(routes);
+  }
+
+  private trackViewTop() {
+    this.viewTop = (this.windowRef.nativeWindow.pageYOffset + this.omnibarService.getHeight());
+    this.documentBottom = Math.round(this.windowRef.nativeWindow.document.documentElement.getBoundingClientRect().bottom);
+  }
+
+  private isCurrent(routes: StacheNavLink[]): void {
+    if (this.scrollEnd()) {
+      routes.forEach((route: StacheNavLink, idx: number) => {
+        route.isCurrent = (idx === (routes.length - 1));
+      });
+      return;
+    }
+
+    routes.forEach((route, index) => {
+      const nextRoute = routes[index + 1];
+      if (nextRoute && nextRoute.offsetTop <= this.viewTop) {
+        route.isCurrent = false;
+        return;
+      }
+
+      route.isCurrent = route.offsetTop <= this.viewTop;
+    });
+  }
+
+  private scrollEnd = () => (this.windowRef.nativeWindow.innerHeight + 5) >= this.documentBottom;
```

Comment:
Can we format this like the rest of the Component pattern?  Not just a variable pointing to an anonymous function?